src/parser/jobs/mnk/BuffUptime.js

```diff
@@ -0,0 +1,161 @@
+
+		// Apparently DK is a bit busted but let's use the debuff for now
+		this.addHook(['applybuff', 'refreshbuff'], {by: 'player', abilityId: STATUSES.BLUNT_RESISTANCE_DOWN.id}, this._onDragonKick)
+		this.addHook(['applybuff', 'refreshbuff'], {by: 'player', abilityId: STATUSES.TWIN_SNAKES.id}, this._onTwinSnakes)
+
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onCast(event) {
+		const action = getAction(event.ability.guid)
+
+		if (!action) {
+			return
+		}
+
+		if (action.onGcd) {
+			this._gcdsSinceDK[event.targetID] += 1
+			this._gcdsSinceTS++
+		}
+	}
+
+	_onDragonKick(event) {
+		// Make sure we're tracking for this target
+		const lastApplication = this._lastDragonKickUse[event.targetID]
+
+		// If it's not been applied yet set it and skip out
+		if (!lastApplication) {
+			this._lastDragonKickUse[event.targetID] = event.timestamp
+			return
+		}
+
+		// This logic is busted if the player's GCD is under 1.67s
+		if (this._gcdsSinceDK[event.targetID] < GCD_CYCLE_LENGTH) {
+			this._earlyDragonKicks++
+		}
+
+		this._lastDragonKickUse[event.targetID] = event.timestamp
+		this._gcdsSinceDK[event.targetID] = 0
+	}
+
+	_onTwinSnakes(event) {
+		// If it's not been applied yet set it and skip out
+		if (!this._lastTwinSnakesUse) {
+			this._lastTwinSnakesUse = event.timestamp
+			return
+		}
+
+		if (this._gcdsSinceTS < GCD_CYCLE_LENGTH) {
+			this._earlyTwinSnakes++
+		}
+
+		this._lastTwinSnakesUse = event.timestamp
+		this._gcdsSinceTS = 0
+	}
+
+	_onComplete() {
+		this.checklist.add(new Rule({
+			name: 'Keep Dragon Kick up',
+			description: <Fragment>
+				Dragon Kick's blunt resistance debuff should always be applied to your primary target.
+			</Fragment>,
+			requirements: [
+				new Requirement({
+					name: <Fragment><ActionLink {...ACTIONS.DRAGON_KICK} /> uptime</Fragment>,
+					percent: () => this.getDebuffUptimePercent(STATUSES.BLUNT_RESISTANCE_DOWN.id),
+				}),
+			],
+		}))
+
+		this.checklist.add(new Rule({
+			name: 'Keep Twin Snakes up',
+			description: <Fragment>
+				Twin Snakes is an easy 10% buff to your DPS across the board.
+			</Fragment>,
+			requirements: [
+				new Requirement({
+					name: <Fragment><ActionLink {...ACTIONS.TWIN_SNAKES} /> uptime</Fragment>,
+					percent: () => this.getBuffUptimePercent(STATUSES.TWIN_SNAKES.id),
+				}),
+			],
+		}))
+
+		if (this._earlyDragonKicks) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.DRAGON_KICK.icon,
+				content: <Fragment>
+					Avoid refreshing <ActionLink {...ACTIONS.DRAGON_KICK} /> signficantly before its expiration as you're trading it for the higher damage <ActionLink {...ACTIONS.BOOTSHINE} />.
+				</Fragment>,
+				severity: SEVERITY.MEDIUM,
+				why: <Fragment>
+					{this._earlyDragonKicks} rotation cycles were interrupted by early refreshes.
+				</Fragment>,
+			}))
+		}
+
+		if (this._earlyTwinSnakes) {
+			const _lostTruePotency = this._earlyTwinSnakes * (ACTIONS.TRUE_STRIKE.potency - ACTIONS.TWIN_SNAKES.potency)
```

Comment:
Minor, but this doesn't need to be marked as private since it's local.

Reply:
Is that a thing? Most other modules seem to just yolo consts so I have no idea.